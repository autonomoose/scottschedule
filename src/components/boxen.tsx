/*
    various boxes
*/
import React from 'react';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';

interface iCaptionBox {
    caption: string,
    color?: string,
    xpad?: string, // rem
}
// quick caption box
export const CaptionBox = (props: iCaptionBox) => {
    const { caption, color, xpad } = props;
    const wkColor = color || 'inherit';
    const wkXpad = xpad || '0.5rem';
    return (
      <Box px={wkXpad} mt={-1.75}>
        <Typography variant='caption' color={wkColor}
          sx={{ bgcolor:'background.paper', position: 'relative', zIndex: '2'}} >
          &nbsp;{caption}&nbsp;
        </Typography>
      </Box>
)}

