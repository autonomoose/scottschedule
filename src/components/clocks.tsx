// more clocks
//   - clocks must avoid hooks, clocks are conditionals
//   - define new clock and register it with widget

import React from 'react';

import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

// use this as an outside interface to clocks
interface ClockWidgetProps {
  currClock: string,
  onComplete: (status: string) => void,
  currSched?: string,
  descrSched?: string,
}
export const ClockWidget = (props: ClockWidgetProps) => {
    return(
      <Box display='flex' flexDirection='column'>
      {
        {
          'digital1': <ClockDigital1 onComplete={props.onComplete} />,
          'digital2': <ClockDigital2 onComplete={props.onComplete} />,
        }[props.currClock]
      }
      </Box>
)}

interface ClockProps {
  onComplete: (status: string) => void,
}
export const ClockDigital2 = (props: ClockProps) => {
    return(
      <>
        <Button data-testid='change clock' sx={{margin: 0}} onClick={() => props.onComplete('scheduler')}>
          <Typography variant='h1' id='mainclock' data-testid='mainclock'
            sx={{fontSize:150, fontWeight: 600, padding: 0, margin: 0}}>00:00</Typography>
        </Button>
        <Typography variant='subtitle1' id='mainpm' data-testid='mainpm' sx={{fontSize:20, fontWeight: 600}}>
          PM
        </Typography>
        <Box mx={4} display='flex' justifyContent='space-between'>
          <Typography mx={1} variant='h4' id='maindate' data-testid='maindate'>Day, 01/01</Typography>
          <Button onClick={() => props.onComplete('close')}>Scheduler</Button>
        </Box>
      </>
) };

export const ClockDigital1 = (props: ClockProps) => {
    return(
      <>
        <Button data-testid='change clock' sx={{margin: 0, padding: 0}} onClick={() => props.onComplete('digital2')}>
          <Box display='flex' justifyContent='flex-start' alignItems='stretch'>
            <Box mt={1} display='flex' alignItems='flex-start'>
              <Typography id='mainpm' data-testid='mainpm'
               sx={{lineHeight:1, fontSize:20, fontWeight: 600}}>
                PM
              </Typography>
            </Box>
            <Typography id='compclock' data-testid='compclock'
             sx={{marginLeft: '-2px', letterSpacing: '-30px', lineHeight:.8, fontSize:230, fontWeight: 600, padding: 0, margin: 0}}>
               0
            </Typography>
            <Box ml={3} mt={1}>
              <Typography id='compminutes' data-testid='compminutes'
               sx={{letterSpacing: '-4px',lineHeight:.92, fontSize:150, fontWeight: 800, padding: 0, margin: 0}}>
                00
              </Typography>
              <Typography id='maindate' data-testid='maindate'
               sx={{ textTransform: 'none', lineHeight:1, fontSize:26, padding: 0, margin: 0}}>
                Day, 01/01
              </Typography>
            </Box>
           </Box>
        </Button>
        <Box mx={4} display='flex' justifyContent='flex-end'>
          <Button onClick={() => props.onComplete('digital1-color')}>Color</Button>
          <Button onClick={() => props.onComplete('close')}>Scheduler</Button>
        </Box>
      </>
) };



