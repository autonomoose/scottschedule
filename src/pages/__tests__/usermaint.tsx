/*
usermaint
 contains mocks for userutil components
 - MockEmailChgDialog (MockEmailChgDialogProps)
 - PasswordChgDialog
 - UserDelDialog
*/
import React from "react";
import { render, waitFor } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import "@testing-library/jest-dom";

import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import UserMaintPage from "../usermaint";
import { EmailChgVerDialog, PasswordChgDialog, UserDelDialog } from '../../components/userutil';

import { Auth } from "aws-amplify"
jest.mock('aws-amplify');

interface MockEmailChgDialogProps {
    dialogOpen: boolean,
    dialogClose: (arg0:string) => void,
}
const MockEmailChgDialog = (props: MockEmailChgDialogProps) => {
    const {dialogOpen, dialogClose} = props;
    const handleCancel = () => {
        dialogClose('');
    };
    const handleChanged = () => {
        dialogClose('email');
    };
    const handleVerify = () => {
        dialogClose('ver');
    };
    return(
        <Dialog onClose={handleCancel} aria-labelledby="namedialog-title" open={dialogOpen}>
        <div>Hidden dialog
          { (dialogOpen) &&
              <>
              <div data-testid="mockdialog">Open</div>
              <Button onClick={handleCancel} color="secondary" variant="outlined">
                Dialog Cancel
              </Button>
              <Button onClick={handleChanged} color="secondary" variant="outlined">
                Dialog Change
              </Button>
              <Button onClick={handleVerify} color="secondary" variant="outlined">
                Dialog Verify
              </Button>
              </>
          }
        </div>
        </Dialog>
        );
};

jest.mock('../../components/userutil', () => ({
    __esModule: true,
    default: jest.fn(),
    EmailChgVerDialog: jest.fn(),
    PasswordChgDialog: jest.fn(),
    UserDelDialog: jest.fn(),
}));

beforeAll(() => {
  (EmailChgVerDialog as jest.Mock).mockImplementation(MockEmailChgDialog);
  (PasswordChgDialog as jest.Mock).mockImplementation(MockEmailChgDialog);
  (UserDelDialog as jest.Mock).mockImplementation(MockEmailChgDialog);
});


const mockEnqueue = jest.fn();
jest.mock('notistack', () => ({
  ...jest.requireActual('notistack'),
  useSnackbar: () => {
    return {
      enqueueSnackbar: mockEnqueue
    };
  }
}));

// local test helpers
//
const mytest = <UserMaintPage />;
const mySetup = async () => {
    const utils = render(mytest);
    await waitFor(() => {
        expect(utils.getByTestId('mainPageDisplay')).toBeVisible();
    });
    const chgEmailButton = utils.getByRole('button', {name: /change email/i});
    const chgPwdButton = utils.getByRole('button', {name: /change password/i});
    const prevAuthAttrib = Auth.userAttributes;

    return {
        ...utils,
        chgEmailButton,
        chgPwdButton,
        prevAuthAttrib,
    }
}

describe("UserMaintPage", () => {
  it("renders snapshot correctly", async () => {
    const {asFragment, getByTestId} = render(mytest);
    await waitFor(() => {
        expect(getByTestId('mainuser')).toBeVisible();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it("opens and closes change dialogs as primary", async () => {
    const utils = await mySetup();

    // open change password and cancel
    expect(utils.chgPwdButton).toBeEnabled();
    userEvent.click(utils.chgPwdButton);
    await waitFor(() => {
        expect(utils.getByTestId('mockdialog')).toBeVisible();
    });
    userEvent.click(utils.getByRole('button', {name: /cancel/i}));
    await waitFor(() => {
        utils.getByRole('main');
    });

    // open change email and cancel
    expect(utils.chgEmailButton).toBeEnabled();
    userEvent.click(utils.chgEmailButton);
    await waitFor(() => {
        expect(utils.getByTestId('mockdialog')).toBeVisible();
    });
    userEvent.click(utils.getByRole('button', {name: /cancel/i}));
    await waitFor(() => {
        utils.getByRole('main');
    });

    // click on close account and then cancel
    userEvent.click(utils.getByRole('button', {name: /close account/i}));
    await waitFor(() => {
        expect(utils.getByTestId('mockdialog')).toBeVisible();
    });

    userEvent.click(utils.getByRole('button', {name: /cancel/i}));
    await waitFor(() => {
        utils.getByRole('main');
    });

  }, 60000);

  it("checks change email submit", async () => {
    const utils = await mySetup();

    // open change email and click change
    expect(utils.chgEmailButton).toBeEnabled();
    userEvent.click(utils.chgEmailButton);
    await waitFor(() => {
        expect(utils.getByTestId('mockdialog')).toBeVisible();
    });

    userEvent.click(utils.getByRole('button', {name: /change/i}));
    await waitFor(() => {
        utils.getByRole('main');
    });

  }, 30000);

  it("checks verify email submit", async () => {
    // @ts-expect-error: until aws-amplify gets formal typing
    Auth.userAttributes = jest.fn(() => Promise.resolve([
        {Name: 'sub', Value: 'test'},
        {Name: 'email_verified', Value: 'false'},
        {Name: 'email', Value: 'testuser1@test.com'},
        {Name: 'custom:tenant', Value: 'test'},
        {Name: 'phone_number_verified', Value: 'false'},
        {Name: 'phone_number', Value: '111'},
      ]));

    const utils = await mySetup();
    // open change email and click verify
    expect(utils.chgEmailButton).toBeEnabled();
    userEvent.click(utils.chgEmailButton);
    await waitFor(() => {
        expect(utils.getByTestId('mockdialog')).toBeVisible();
    });

    userEvent.click(utils.getByRole('button', {name: /verify/i}));
    await waitFor(() => {
        utils.getByRole('main');
    });

    Auth.userAttributes = utils.prevAuthAttrib;
  }, 30000);

  it("checks re-verify email submit", async () => {
    const prevAuthAttrib = Auth.userAttributes;
    // @ts-expect-error: until aws-amplify gets formal typing
    Auth.userAttributes = jest.fn(() => Promise.resolve([
        {Name: 'sub', Value: 'test'},
        {Name: 'email_verified', Value: 'false'},
        {Name: 'email', Value: 'testuser1@test.com'},
        {Name: 'custom:tenant', Value: 'test'},
        {Name: 'phone_number_verified', Value: 'false'},
        {Name: 'phone_number', Value: '111'},
      ]));

    const utils = await mySetup();
    // open change email and click change
    const verButton = utils.getByRole('button', {name: /verify email/i});
    userEvent.click(verButton);
    await waitFor(() => {
        expect(utils.getByTestId('mockdialog')).toBeVisible();
    });

    userEvent.click(utils.getByRole('button', {name: /change/i}));
    await waitFor(() => {
        utils.getByRole('main');
    });

    Auth.userAttributes = prevAuthAttrib;
  }, 15000);

  it("opens and closes del user as secondary", async () => {
    // @ts-expect-error: until aws-amplify gets formal typing
    Auth.userAttributes = jest.fn(() => Promise.resolve([
      {Name: 'sub', Value: 'test'},
      {Name: 'email_verified', Value: 'true'},
      {Name: 'email', Value: 'testuser1@test.com'},
      {Name: 'custom:tenant', Value: 'different'},
    ]));

    const utils = await mySetup();

    const delUserButton = utils.getByRole('button', {name: /delete user/i});
    expect(delUserButton).toBeEnabled();
    userEvent.click(delUserButton);
    await waitFor(() => {
        expect(utils.getByTestId('mockdialog')).toBeVisible();
    });

    userEvent.click(utils.getByRole('button', {name: /cancel/i}));
    await waitFor(() => {
        utils.getByRole('main');
    });

    Auth.userAttributes = utils.prevAuthAttrib;
  }, 15000);

  it("handles a thrown error on userAttributes", async () => {
    Auth.userAttributes = jest.fn(() => Promise.reject('mockfailed userAttribute'));

    const consoleWarnFn = jest.spyOn(console, 'warn').mockImplementation(() => jest.fn());
    const utils = await mySetup();

    Auth.userAttributes = utils.prevAuthAttrib;

    expect(consoleWarnFn).toHaveBeenCalledTimes(1);
    consoleWarnFn.mockRestore();
  }, 10000);

});
