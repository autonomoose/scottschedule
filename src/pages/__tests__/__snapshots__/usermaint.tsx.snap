// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMaintPage renders snapshot correctly 1`] = `
<div
  data-testid="authenticator"
>
   
  <div
    style="margin: 1rem auto; min-height: 100vh; text-align: center;"
  >
    <div
      class="MuiBox-root css-0"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionFixed mui-fixed css-1bzzxac-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ti0wu3-MuiToolbar-root"
        >
          <button
            aria-label="Open menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1t7w49f-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
            href="/home"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <img
                alt="Werner Digital"
                src="../images/wernerdigital-hosted.png"
                width="26"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-4peds9-MuiTypography-root"
              >
                Scottschedule
              </p>
            </div>
          </a>
          <div
            class="MuiBox-root css-yukqif"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/usermaint"
              style="margin-right: 1rem;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                mockuname
              </p>
            </a>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1vbvugf-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Sign Out
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <button
            aria-label="Dark/light mode"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1t7w49f-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LightModeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41l-1.06-1.06zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
      <div
        class="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-1hrb81m-MuiPaper-root-MuiDrawer-paper"
          style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
        >
          <div
            class="MuiBox-root css-1ecnymn"
          >
            <div
              class="MuiBox-root css-xwg089"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                MainÂ Menu
              </h6>
              <button
                aria-label="Close menu"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1drgtl0-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropUpIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m7 14 5-5 5 5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
          />
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/home"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="HomeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Home
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/usermaint"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Account
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/scheds"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Schedules
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/events"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Events
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/help"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Help
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <div
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Logout
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-h8m1q6"
    >
      <main>
        <div
          data-testid="mainPageDisplay"
          style="text-align: left;"
        >
          <div
            class="MuiBox-root css-7kebnq"
          >
            <div
              style="float: left;"
            >
              <nav
                aria-label="Breadcrumb"
                class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1kkz7jp-MuiTypography-root-MuiBreadcrumbs-root"
              >
                <ol
                  class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
                      href="/home"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    &gt;
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <span>
                      User Settings
                    </span>
                  </li>
                </ol>
              </nav>
            </div>
            <div>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
                href="/help/usermaint"
              >
                Help
              </a>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-g0zzsc-MuiPaper-root"
            data-testid="mainuser"
            style="margin: 0px; padding: 1px; clear: both;"
          >
            <div
              class="MuiBox-root css-164r41r"
            >
              <div
                class="MuiBox-root css-k008qs"
              >
                <h3>
                  Welcome
                </h3>
                <div
                  class="MuiBox-root css-6ofyve"
                >
                  to the User Settings page.  If you need to contact support, please include the following:
                </div>
              </div>
              <div
                class="MuiBox-root css-jltqb6"
              >
                <ul>
                  <li>
                     Account key:
                    <br />
                     
                    test
                     
                  </li>
                  <li>
                     User key:
                    <br />
                     
                    test
                     
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-tvv90y"
          >
            <div
              class="MuiBox-root css-gz1xpb"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gyd2pn-MuiPaper-root-MuiCard-root"
              >
                <div
                  class="MuiBox-root css-1impqaw"
                >
                  <button
                    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1is3qj5-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Change Password 
                  </button>
                </div>
                <div
                  class="MuiBox-root css-18vgi6o"
                >
                  Good account security starts with strong passwords that are changed regularly. 
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-gz1xpb"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gyd2pn-MuiPaper-root-MuiCard-root"
              >
                <div
                  class="MuiBox-root css-1impqaw"
                >
                  <button
                    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1is3qj5-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Change Email 
                  </button>
                </div>
                <div
                  class="MuiBox-root css-18vgi6o"
                >
                  <strong>
                    testuser1@test.com
                  </strong>
                  <div
                    class="MuiBox-root css-1l4w6pd"
                  >
                    <span>
                      (verified)
                    </span>
                  </div>
                  <p>
                    It is important to have a valid email address for account security and password recovery.
                  </p>
                  <p>
                    We verify new email addresses by sending a code to that address.
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-gz1xpb"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gyd2pn-MuiPaper-root-MuiCard-root"
              >
                <div
                  class="MuiBox-root css-1impqaw"
                >
                  <button
                    class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-bt5zmn-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Close Account 
                  </button>
                </div>
                <div
                  class="MuiBox-root css-18vgi6o"
                >
                  This will close your account and log you off. Any account data will be securely deleted within 7 days of account closing date.
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
    <footer
      style="padding-top: 40px;"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1bnxa12"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/home"
        >
          Home
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/usermaint"
        >
          Account
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/scheds"
        >
          Schedules
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/events"
        >
          Events
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/help"
        >
          Help
        </a>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-3merus-MuiTypography-root"
      >
        Â© 2021-
        2022
        , Werner Digital Technology Inc
      </span>
    </footer>
  </div>
</div>
`;
